connectors/src/connectors/zendesk/index.ts

```diff
@@ -170,102 +161,106 @@ export class ZendeskConnectorManager extends BaseConnectorManager<null> {
+            }
           }
-        }
-        if (permission === "read") {
-          const newBrand = await allowSyncZendeskTickets({
-            connectorId: this.connectorId,
-            connectionId,
-            brandId: brandTicketsId,
-            subdomain,
-          });
-          if (newBrand) {
-            toBeSignaledTicketsIds.add(newBrand.brandId);
+          if (permission === "read") {
+            const newBrand = await allowSyncZendeskHelpCenter({
+              connectorId,
+              connectionId,
+              brandId: objectId,
+              subdomain,
+            });
+            if (newBrand) {
+              toBeSignaledHelpCenterIds.add(newBrand.brandId);
+            }
           }
+          break;
         }
-      } else if (categoryId) {
-        if (permission === "none") {
-          const revokedCategory = await revokeSyncZendeskCategory({
-            connectorId: this.connectorId,
-            categoryId,
-          });
-          if (revokedCategory) {
-            toBeSignaledCategoryIds.add(revokedCategory.categoryId);
+        case "tickets": {
+          if (permission === "none") {
+            const revokedCollection = await revokeSyncZendeskTickets({
+              connectorId,
+              brandId: objectId,
+            });
+            if (revokedCollection) {
+              toBeSignaledTicketsIds.add(revokedCollection.brandId);
+            }
+          }
+          if (permission === "read") {
+            const newBrand = await allowSyncZendeskTickets({
+              connectorId,
+              connectionId,
+              brandId: objectId,
+              subdomain,
+            });
+            if (newBrand) {
+              toBeSignaledTicketsIds.add(newBrand.brandId);
+            }
           }
+          break;
         }
-        if (permission === "read") {
-          const newCategory = await allowSyncZendeskCategory({
-            subdomain,
-            connectorId: this.connectorId,
-            connectionId,
-            categoryId,
-          });
-          if (newCategory) {
-            toBeSignaledCategoryIds.add(newCategory.categoryId);
+        case "category": {
+          if (permission === "none") {
+            const revokedCategory = await revokeSyncZendeskCategory({
+              connectorId,
+              categoryId: objectId,
+            });
+            if (revokedCategory) {
+              toBeSignaledCategoryIds.add(revokedCategory.categoryId);
+            }
+          }
+          if (permission === "read") {
+            const newCategory = await allowSyncZendeskCategory({
+              subdomain,
+              connectorId,
+              connectionId,
+              categoryId: objectId,
+            });
+            if (newCategory) {
+              toBeSignaledCategoryIds.add(newCategory.categoryId);
+            }
           }
+          break;
         }
+        // we do not set permissions for single articles and tickets
```

Comment:
in that case maybe send an error if somebody tries to call a setPermissions with another type. Make it explicit and ensure all types are handled by adding an assertNever : 
```
case "article":
case "ticket":
  throw new Error("Unexpected type");
default: 
  assertNever(type);
```